java/core/src/main/java/com/microsoft/bond/protocol/SimpleBinaryWriter.java

```diff
@@ -49,6 +49,13 @@ public void writeVersion() throws IOException {
     @Override
     public void writeFieldOmitted(
             final BondDataType type, final int id, final FieldMetadata metadata) throws IOException {
+        // Simple doesn't support omitting fields, so we need to explicitly write the default values
+        // for optional fields.
+        //
+        // Fields that are = nothing have no default, so it is an unrecoverable error to attempt to
+        // serialize such a field if it is actually set to nothing (e.g., a SomethingObject which is
+        // null).
+
```

Comment:
Yeah. My intent was that you would read it as an implementer's note on this function, not on any particular lines within it.